compiler/optimizer/RegisterCandidate.cpp

```diff
@@ -608,11 +608,12 @@ TR_RegisterCandidate::setWeight(TR::Block * * blocks, int32_t *blockStructureWei
        //
       if (extendedBlockFreq <= blockFreq)
          {
-         if ((uint32_t)blockWeight > (*_loadsAndStores)[extendedBlock->getNumber()])
-            (*_loadsAndStores)[extendedBlock->getNumber()] = blockWeight;
+         if ((uint32_t)blockWeight > _blocks.getNumberOfLoadsAndStores(extendedBlock->getNumber()))//(*_loadsAndStores)[extendedBlock->getNumber()])
+            //(*_loadsAndStores)[extendedBlock->getNumber()] = blockWeight;
+            _blocks.setNumberOfLoadsAndStores(extendedBlock->getNumber(), blockWeight);
          }
-      if ((uint32_t)blockWeight > (*_loadsAndStores)[blockNumber])
-         (*_loadsAndStores)[blockNumber] = blockWeight;
+      //if ((uint32_t)blockWeight > (*_loadsAndStores)[blockNumber])
+      //   (*_loadsAndStores)[blockNumber] = blockWeight;
```

Comment:
Is this change in logic intentional ? The thrust of most of the other changes in this commit are to keep existing behaviour.